client/src/Pages/Categories.js

```diff
@@ -28,16 +28,12 @@ function Categories({ match }) {
     setLoading(true);
     getAll(1, currentCategory, query)
       .then((res) => {
-        if (query === "") {
```

Comment:
The reported issue with duplicates was due to this logic here. This wasn't necessary because the server supports queries and categories. We don't need front end logic when we can get the data directly from the server.

Note: once this issue was resolved it exposed a second issue: searching for a query while filtering to a category doesn't narrow results to the category it only returns the query and disregards the category. I would recommend refactoring the getAll function to support querying while filtering for a better user experience at scale (hundreds, thousands of products).